framework/include/actions/CreateApplicationBlockAction.h

```diff
@@ -0,0 +1,32 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "Action.h"
+
+/**
+ * Action that create Application Block. It is noted that most information has already been
+ * parsed before the MOOSE object is built. This action mainly serve the purpose of completing
+ * the action system workflow for MOOSE objects.
+ */
+class CreateApplicationBlockAction : public Action
```

Comment:
Do we need this object? Or can we just tell moose to completely ignore `[Application]` and get it all from the parser?

Reply:
If we add a parameter like 'command_line_args' or 'min_ad_size', will we use this action or will we create new actions for each independent parameter ? @loganharbour

Reply:
let s talk about it tomorrow?